src/Scope.agda

```diff
@@ -0,0 +1,222 @@
+-- An abstract interface to representations of scopes
+
+{- Design constraints:
+- Abstract interface for working with scopes
+- Well-scoped syntax
+- Scopes are runtime-irrelevant
+- Joining two scopes α and β should only increase the size of the indices by a constant
+- Possibility to construct from a proof of x ∈ α a "remainder scope" α' with x removed
+- Empty scopes should not matter (see below)
+- Variable indices are somehow bounded in size by the size of the scope
+  (todo: add a proper notion of "size" to the interface?)
+-}
+
+{- Problem: should empty scopes matter?
+
+We say that "empty scopes matter" if the following are NOT all equivalent:
+- α ⋈ β ≡ γ
+- (∅ <> α) ⋈ β ≡ γ
+- (α <> ∅) ⋈ β ≡ γ
+- α ⋈ (∅ <> β) ≡ γ
+- α ⋈ (β <> ∅) ≡ γ
+- α ⋈ β ≡ (∅ <> γ)
+- α ⋈ β ≡ (γ <> ∅)
+
+It is clearly desirable to have empty scopes not matter, but designing a
+concrete implementation that satisfies this (as well as the other properties
+above) is challenging.
+
+-}
+
+{-# OPTIONS --no-forcing #-} -- temporary until #6867 is fixed
+
+-- open import Utils
+open import Haskell.Prelude hiding (All; _∘_)
+open import Haskell.Law.Equality
+import Haskell.Law.List as List
+
+open import Utils.Erase
+open import Utils.Tactics
+open import Utils.Dec as Dec
+import Utils.List as List
+
+module Scope where
+
+  ----------------------------
+
+  @0 <>-∅ : {α : Scope name} → α <> empty ≡ α
```

Comment:
Uhm I'm confused, how is this the first definition in this module? Should this module even still exist?

Reply:
This module contains what I haven't yet moved to somewhere else in `Scope/`. It no longer typechecks and will get removed.